File: 신고결과받기.py
def solution(id_list, report, k):

    # 한번에 한 명의 유저
    # k번 이상 = 정지, 리폿 먹인 사람은 정지사실 통보받음. 이 과정은 마지막에 한꺼번에 이루어짐
    # answer = 각 유저가 받은 메일의 수
    # Pseudo code
    # report를 for문으로 돌면서 report[i][1]의 count가 k가 되면 밴이 된다. 그 경우
    # id_list에서 report[i][0]에 해당하는 이름과 같은 인덱스를 result의 인덱스에 1을 추가
    stop = []
    answer = [0] * len(id_list)
    dicReports = {id: [] for id in id_list}
    for i in set(report):
        report = i.split(' ')
        stop.append(report[1])
        dicReports[report[0]].append(report[1])

    stop = set([i for i in stop if stop.count(i) >= k])

    for key, value in dicReports.items():
        for s in stop:
            if s in value:
                answer[id_list.index(key)] += 1
    return answer

File: 신고결과받기.js
function solution(id_list, report, k) {
  let newReport = report.filter((el, idx) => {
    return report.indexOf(el) === idx;
  });

  let answer = Array.from({ length: id_list.length }, () => 0);
  let reported = newReport.map((el) => el.split(" ")[1]);

  let map = {};
  id_list.map((id) => {
    map[id] = 0;
  });

  for (let i = 0; i < reported.length; i++) {
    if (map[reported[i]] > 0) {
      map[reported[i]] += 1;
    } else {
      map[reported[i]] = 1;
    }
  }

  for (let key in map) {
    if (map[key] >= k) {
      //만약 report 각요소의 [1]요소가 key일 경우 그 [0]번째 요소에 해당하는 id_list의 idx값을 추출해 answer[idx]값에 1을 더한다.
      report.map((el) => {
        if (el.split(" ")[1] === key) {
          answer[id_list.indexOf(el.split(" ")[0])] += 1;
        }
      });
    }
  }
  return answer;
}

id_list = ["muzi", "frodo", "apeach", "neo"];
report = ["muzi frodo", "apeach frodo", "frodo neo", "muzi neo", "apeach muzi"];
console.log(solution(id_list, report, 2));

//간단한 코드를 찾아볼 수 있으니 프로그래머스 참고
